### ISSUE TITLE
Explain Predictions not working

### ISSUE THREAD

**Issue by xnmp** [TeamHG-Memex/eli5#214]
So I have a model called `gbm`. `gbm.predict_proba(X)` works, and `explain_weights(gbm)` works. But when I try to write 

```
explain_prediction(gbm, X.iloc[1])
```
I get `ValueError: DataFrame.dtypes for data must be int, float or bool. Did not expect that data types in fields (every field)`.

If I instead write 

```
explain_prediction(gbm, X.iloc[1].values)
```

I get an answer, but all the features are label `x1`,`x2`,`x3`, etc. 

Any idea what's going on?

**Comment by lopuhin** [comment]
hi @xnmp ! Can you please give the original traceback you get? I tried to reproduce the issue, but didn't get an error - probably you are doing something differently: 
```import eli5
import pandas as pd
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.datasets import load_iris

data = load_iris()
data.keys()

X = pd.DataFrame(data['data'], columns=data['feature_names'])

gbm = GradientBoostingClassifier()
gbm.fit(X, data['target'])

eli5.show_weights(gbm)
eli5.show_prediction(gbm, X.iloc[1])
```
![image](https://user-images.githubusercontent.com/424613/27174848-2891cdea-51c6-11e7-8839-56034849144c.png)

**Comment by lextoumbourou** [comment]
I'm having the same issue with a LightGBM. I found a simple workaround, which is to specify the columns using the  `feature_names` arg:

```
explain_prediction(gbm, X.iloc[1].values, feature_names=list(X.columns))
```

**Comment by agnesvanbelle** [comment]
I had the same problem and above did not solve it for me. What worked for me was using `X.iloc[[1]]` instead of `X.iloc[1]`.  The latter form automatically converts a row to a series, which converts the datatypes in the row to "object" type if they differ.
